### PR TITLE
Add two-stage pretrain→finetune pipeline with Optuna tuning

### PR DESCRIPTION
**Pull request by karnamgyal**
The issue #57 is not complete yet, but this is a great place for everyone to work off of. 

## What I added

- Two-stage training: pretrain on healthy data, finetune on impaired (--stage pretrain | finetune | direct in training/train.py)
- Works with LSTM, CNN, and GNN (--model lstm | cnn | gnn)
- scripts/run_two_stage.py — runs pretrain → finetune in one go
- scripts/tune_model.py — Optuna hyperparameter search (--both-stages searches pretrain + finetune)
- Healthy/impaired data splits in loaders + --separate-arm-dirs preprocess
- results/README.md with metrics; plots in results/best_pretrain/ and results/best_finetune/ (checkpoints stay local, gitignored)

- I obtained a CNN with 80% final per finger test accuracy, it's metrics should be found in the results/best_finetuned folder. LSTM did not improve much yet. 

## How to run

1. Put data from the mega link in the correct folder (processed and raw must be in the datasets directory).

2. Preprocess:
   python scripts/run_preprocess.py --separate-arm-dirs

3. Two-stage pipeline (swap --model for lstm / cnn / gnn):
   python scripts/run_two_stage.py --model cnn --device cuda --skip-preprocess

4. Single stage (direct, pretrain, or finetune only):
   python training/train.py --model lstm --stage direct --device cuda
   python training/train.py --model cnn --stage pretrain --device cuda
   python training/train.py --model cnn --stage finetune --pretrained-ckpt results/best_pretrain/cnn/checkpoint_best.pt --device cuda

5. Tune hyperparameters (Optuna):
   python scripts/tune_model.py --model cnn --both-stages --n-trials 30 --device cuda

Best Optuna configs: training/tuning/{cnn,lstm}/both_stages_summary.json
Checkpoints save to: results/best_pretrain/{model}/ and results/best_finetune/{model}/ after training.

### CODE REVIEW

**Review by Divy-1729 — APPROVED** [review]
LGTM. Great work!
